src/app/sidebar/Sidebar.tsx

```diff
@@ -1,24 +1,68 @@
+import { Flex, Heading } from '@chakra-ui/react';
 import React from 'react';
+import InfiniteScroll from 'react-infinite-scroll-component';
+import { Card } from './components/Card';
+import { CardDropDown } from './components/CardDropDown';
+import { Course, Term, useGetCourses, useSubjects } from '../../fetchers'
+
 // import StyledHeader from './Header.styles';
 
 
 export interface SidebarProps {
   /**
-   * Content
-   * Subject to change
+   * Term Selected
+   * Determines what term the subjects and courses are from
    */
-  content?: string;
+  term: Term;
 }
 
-/**
- * Primary UI component for content
- */
-export const Sidebar: React.FC<SidebarProps> = ({
-  content
-}) => {
+export function Sidebar({ term }: SidebarProps): JSX.Element {
+
+  const { data: subjects, loading: loadingSubjects, error: errorSubjects } = useSubjects({ term: term });
+  const { data: courses, loading: loadingCourses, error: errorCourses } = useGetCourses({ term: term })
+
+  const parsedCourses: { [subject: string]: Course[] } = {}
+
+  if (courses != null) {
+    for (let i = 0; i < courses.length; i++) {
+      let course: Course = courses[i];
+      let subject = course.subject;
+      if (!(subject in parsedCourses)) {
+        parsedCourses[subject] = [];
+      }
+      parsedCourses[subject].push(course)
+    }
+  }
```

Comment:
Some might argue which is better but.
```ts
  const p = subjects?.reduce(
    (acc, curr) => ({
      ...acc,
      [curr.subject]: courses?.filter((c) => c.subject === curr.subject),
    }),
    {} as { [subject: string]: Course[] | undefined }
  );
  ```
  Not sure which is better give how it turns out.

Reply:
When using filter, it calls for every course. So, I combined reduce and previous logic to make it more efficient.